### ISSUE TITLE
Layout/SpaceInsideArrayLiteralBrackets cop cannot handle an array literal as an argument after a heredoc

### ISSUE THREAD

**Issue by timrogers** [bbatsov/rubocop#5646]
The `Layout/SpaceInsideArrayLiteralBrackets` fails to inspect a file which contains an array literal as an argument after a heredoc is started:

```ruby
ActiveRecord::Base.connection.execute(<<-SQL, [self.class.to_s]).first["count"]
SELECT COUNT(widgets.id) FROM widgets
WHERE widget_type = $1
SQL
```

The code involved looks very odd, but it is valid and documented Ruby syntax - see the [docs](https://docs.ruby-lang.org/en/2.5.0/syntax/literals_rdoc.html), scrolling down to the code sample introduced with "To call a method on a heredoc place it after the opening identifier:".

--------

## Expected behavior

Rubocop should successfully inspect this code, and not flag any offences (since we have the cop configured with the `EnforcedStyle` `no_space`).

## Actual behavior

You get a message like the following:

```
An error occurred while Layout/SpaceInsideArrayLiteralBrackets cop was inspecting /Users/timrogers/Projects/<redacted>.rb:45:24.
```

The complete backtrace looks like this:

```
Scanning /Users/timrogers/Projects/<redacted>.rb
An error occurred while Layout/SpaceInsideArrayLiteralBrackets cop was inspecting /Users/timrogers/Projects/<redacted>:45:24.
undefined method `+' for nil:NilClass
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/cop/layout/space_inside_array_literal_brackets.rb:122:in `next_to_newline?'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/cop/layout/space_inside_array_literal_brackets.rb:79:in `on_array'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/cop/commissioner.rb:44:in `block (2 levels) in on_array'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/cop/commissioner.rb:109:in `with_cop_error_handling'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/cop/commissioner.rb:43:in `block in on_array'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/cop/commissioner.rb:42:in `each'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/cop/commissioner.rb:42:in `on_array'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/ast/traversal.rb:101:in `block in on_send'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/ast/traversal.rb:99:in `each'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/ast/traversal.rb:99:in `each_with_index'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/ast/traversal.rb:99:in `on_send'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/cop/commissioner.rb:48:in `on_send'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/ast/traversal.rb:101:in `block in on_send'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/ast/traversal.rb:99:in `each'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/ast/traversal.rb:99:in `each_with_index'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/ast/traversal.rb:99:in `on_send'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/cop/commissioner.rb:48:in `on_send'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/ast/traversal.rb:101:in `block in on_send'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/ast/traversal.rb:99:in `each'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/ast/traversal.rb:99:in `each_with_index'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/ast/traversal.rb:99:in `on_send'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/cop/commissioner.rb:48:in `on_send'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/ast/traversal.rb:95:in `on_def'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/cop/commissioner.rb:48:in `on_def'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/ast/traversal.rb:49:in `block in on_begin'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/ast/traversal.rb:49:in `each'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/ast/traversal.rb:49:in `on_begin'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/cop/commissioner.rb:48:in `on_begin'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/ast/traversal.rb:141:in `on_while'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/cop/commissioner.rb:48:in `on_module'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/ast/traversal.rb:12:in `walk'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/cop/commissioner.rb:60:in `investigate'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/cop/team.rb:114:in `investigate'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/cop/team.rb:102:in `offenses'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/cop/team.rb:44:in `inspect_file'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/runner.rb:259:in `inspect_file'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/runner.rb:206:in `block in do_inspection_loop'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/runner.rb:238:in `block in iterate_until_no_changes'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/runner.rb:231:in `loop'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/runner.rb:231:in `iterate_until_no_changes'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/runner.rb:202:in `do_inspection_loop'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/runner.rb:111:in `block in file_offenses'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/runner.rb:121:in `file_offense_cache'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/runner.rb:109:in `file_offenses'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/runner.rb:100:in `process_file'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/runner.rb:78:in `block in each_inspected_file'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/runner.rb:75:in `each'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/runner.rb:75:in `reduce'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/runner.rb:75:in `each_inspected_file'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/runner.rb:67:in `inspect_files'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/runner.rb:39:in `run'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/cli.rb:156:in `execute_runner'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/cli.rb:84:in `execute_runners'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/lib/rubocop/cli.rb:41:in `run'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/bin/rubocop:13:in `block in <top (required)>'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/2.5.0/benchmark.rb:308:in `realtime'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.53.0/bin/rubocop:12:in `<top (required)>'
/Users/timrogers/.rbenv/versions/2.5.0/bin/rubocop:23:in `load'
/Users/timrogers/.rbenv/versions/2.5.0/bin/rubocop:23:in `<top (required)>'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/cli/exec.rb:75:in `load'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/cli/exec.rb:75:in `kernel_load'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/cli/exec.rb:28:in `run'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/cli.rb:424:in `exec'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/vendor/thor/lib/thor.rb:387:in `dispatch'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/cli.rb:27:in `dispatch'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/vendor/thor/lib/thor/base.rb:466:in `start'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/cli.rb:18:in `start'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/exe/bundle:30:in `block in <top (required)>'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/friendly_errors.rb:122:in `with_friendly_errors'
/Users/timrogers/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/exe/bundle:22:in `<top (required)>'
/Users/timrogers/.rbenv/versions/2.5.0/bin/bundle:23:in `load'
/Users/timrogers/.rbenv/versions/2.5.0/bin/bundle:23:in `<main>'
```

## Steps to reproduce the problem

Include some code like the following:

```ruby
ActiveRecord::Base.connection.execute(<<-SQL, [self.class.to_s]).first["count"]
SELECT COUNT(widgets.id) FROM widgets
WHERE widget_type = $1
SQL
```

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
$ rubocop -V
0.53.0 (using Parser 2.5.0.3, running on ruby 2.5.0 x86_64-darwin16)
```

**Comment by timrogers** [comment]
This is a duplicate of #5637.
